aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -2874,8 +2874,11 @@ static void lu_solve_looped_magma(const Tensor& b, const Tensor& lu, const Tenso
 static void lu_solve_dispatch(const Tensor& b, const Tensor& lu, const Tensor& pivots) {
   auto batch_size = batchCount(lu);
   auto m = lu.size(-2);
+  auto b1 = b.size(-2);
+  auto b2 = b.size(-1);
+  bool over_magma_dim_limit = m > 1024 || b1 > 1024 || b2 > 1024;  // magma implementation of LU solve cannot handle tensors with dimensions > 1024
 #ifdef USE_CUSOLVER
-  if (batch_size == 1 && m > 512) {
+  if ((batch_size == 1 && m > 512) || (batch_size <= 8 && over_magma_dim_limit)) {
```

Comment:
Where do the conditions `batch_size <= 8` and `batch_size > 8` come from?

Reply:
They were chosen based off the benchmarks shown here https://github.com/pytorch/pytorch/pull/59148#issue-656466648 - cuBLAS if faster for for batch size > 8 and cuSOLVER for <= 8. Should I mention this somewhere in the PR?

Reply:
It's fine as it is, thank you for including it here.

Reply:
I think it'd be good to leave a comment in the code pointing this out.

Reply:
The testing script only considers a single dtype, `float64`, could you please confirm the same outcome for `float32`?

Reply:
@Lezcano added.

@nikitaved thanks, I agree confirming the heuristics for different types is important, but is that not an issue to raise for the original PR which chose them (https://github.com/pytorch/pytorch/pull/59148#issue-656466648)? The only contribution I'm proposing in this PR is directing `lu_solve` to not use MAGMA for large b matrices. Please let me know if you disagree.